neuro_san/internals/chat/data_driven_chat_session.py

```diff
@@ -181,11 +190,56 @@ async def streaming_chat(self, user_input: str,
         :return: Nothing.  Response values are put on a queue whose consumtion is
                 managed by the Iterator aspect of AsyncCollatingQueue on the InvocationContext.
         """
+        # Set up some member variable state so that run_it() can use it
+        self.invocation_context = invocation_context
+        journal: Journal = self.invocation_context.get_journal()
+        self.interceptor = InterceptingJournal(journal, origin=None)
+
+        # Set up the input dictionary that will show up in an Observability/tracing app
+        inputs: Dict[str, Any] = {
+            "user_input": user_input,
+            "sly_data": sly_data,
+            "chat_context": chat_context
+        }
+
+        # Set up configuration for creating the tracing context.
+        # These are the bare minimum required to get output and metadata correct
+        # in the tracing reporting.
+        config: Dict[str, Any] = {
+            "interceptor": self.interceptor,
+            "invocation_context": self.invocation_context
+        }
+
+        # Get a toolkit-specific implementation for creating the tracing context
+        tracing_factory: ContextTypeTracingContextFactory = \
+            MasterTracingContextFactory.create_tracing_context_factory()
+        # For the factory args, we are our own run_target.
+        tracing_context: RunTarget = tracing_factory.create_tracing_context(config, run_target=self)
+
+        # Run the run_target that was given back by the factory.
+        await tracing_context.run_it(inputs)
```

Comment:
Since streaming_chat() is the main entry point from the DirectAgentSession stuff,
we subvert what had been going on to set up a tracing context, and push what was under streaming_chat() before into the run_it() method.